conjure/types/conjure_definition.go

```diff
@@ -273,17 +273,23 @@ func (t *namedTypes) resolveType(typeI Type) error {
 			} else {
 				return errors.Errorf("Unresolved optional type reference %s %s", unresolved.Ref.Package, unresolved.Ref.Name)
 			}
+		} else {
```

Comment:
Somewhat of a nit, but think `else if` construct may be better in these cases to make intended control flow clearer